### DISCUSSION TITLE
Repeater clone/duplicate functionality

### DISCUSSION THREAD

**Discussion by klodoma** [directus/directus#17205] [Feature Requests]
I am going to create a fork which creates the "clone" or "duplicate" functionality for a repeater like in Directus 8, because we use it heavily and we need it. 
![image](https://user-images.githubusercontent.com/944350/212871446-69448613-dc31-4467-8883-1901a6357a25.png)

Now, I've already implemented it but I was thinking that maybe it would be better to have a "sub-menu" like for the Fields 
Management? This could be further extended in the future. 

Solution 1:
![image](https://user-images.githubusercontent.com/944350/212871469-5cc846d3-842a-4587-b2d5-22714c08897d.png)

Solution 2:
![image](https://user-images.githubusercontent.com/944350/212872020-2f7f30de-084b-4fc0-a6f0-5fa8c5086177.png)

I think the 2nd solution "more flexible" for future functionality, still the user will have to click twice. 

Before I create a PR I would like to align a bit with your opinion. 

Here are the branches:
https://github.com/klodoma/directus/tree/feature/repeater-clone
https://github.com/klodoma/directus/tree/feature/repeater-clone-submenu

**Comment by paescuj** [comment]
Nice! 🎉 I like the second solution, design-wise and as you mentioned it's more flexible. I don't think it would be an issue that users have to click twice in this case.

Regarding the implementation: I feel like there are existing components (context menu) that could be reused here. But I would have to take a closer look myself...

**Reply by klodoma** [reply]
Thanks for the feedback. Yes, I would also go ahead with the second solution. 

> Regarding the implementation: I feel like there are existing components (context menu) that could be reused here. But I would have to take a closer look myself...

I used the `field-select-menu.vue` as a starting point for the implementation. Source:
https://github.com/directus/directus/blob/main/app/src/modules/settings/routes/data-model/fields/components/field-select-menu.vue

Didn't dig in other places to see if there is more simple approach for the context menu. I'll have a second look. If there is let me know.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
